### DISCUSSION TITLE
Can environment variables be used to configure the server?

### DISCUSSION THREAD

**Discussion by glen-84** [cerbos/cerbos#1701] [Q&A] [answered]
I only see `CERBOS_NO_TELEMETRY`. Is it possible to configure the storage and other values using environment variables?

**Comment by alexolivier** [comment]
Yes you can configure configuration value as an environment variable using

> Config values can reference environment variables by enclosing them between `${}`, for example `${HOME}`. Defaults can be set using `${VAR:default}`.


https://docs.cerbos.dev/cerbos/latest/configuration/index.html

**Reply by glen-84** [reply]
Hi Alex,

I wasn't asking how to reference environment variables in configuration, but how to use environment variables to configure the server.

For example: `CERBOS_STORAGE=git`, `CERBOS_STORAGE_GIT_URL=github.com:cerbos/policy-test.git`, etc.

**Reply by alexolivier** [reply]
Hi,

You would need to create and mount a configuration file with the values you want to set via env variables using the `${}` syntax that way you can define exactly which configuration you set.

Alternatively, you can pass any configuration value in the command line via the `--set=` syntax which references the path to a the configuration value:

```
./cerbos server --config=/path/to/config.yaml --set=server.httpListenAddr=:3592 --set=engine.defaultPolicyVersion=staging
```

**Reply by glen-84** [reply]
Yep, I'm aware of those options, thanks.

I guess the short answer is "No". 🙂

**Reply by alexolivier** [reply]
What is your scenario that makes these options less desirable?

**Reply by glen-84** [reply]
We have a Docker Compose file in our application repository for development purposes, and the existing containers either require no configuration (Redis), or can be configured with environment variables (MySQL).

We wanted to avoid having other (infrastructure-like) files in the repository.

With that said, as there may be some shared configuration between dev and staging/production, it might be best for us to create a custom image with built-in configuration, and instead reference that from the Docker Compose file.

Thanks for your help.

**Reply by charithe** [reply, marked as answer]
Newer versions of Cerbos don't require a config file. So, you can still have a "no-config-file" setup in Docker Compose like this:

```yaml
services:
  cerbos:
    image: ghcr.io/cerbos/cerbos:latest
    command: 
      - "server"
      - "--set=storage.driver=disk"
      - "--set=storage.disk.directory=${XDG_DATA_DIR}/cerbos" 
```

**Reply by glen-84** [reply]
Ah yes, I thought about this option yesterday, and then forgot about it. 🙃

Thanks for the info.
